std/algorithm/searching.d

```diff
@@ -799,15 +802,33 @@ if (isForwardRange!R
                   haystack.popFront();
             }
         }
+        //Because of @@@8804@@@: Avoids both "unreachable code" or "no return statement"
+        static if (isInfinite!R) assert(0);
+        else return -1;
     }
     else
     {
+        static struct Result
+        {
+            ptrdiff_t steps, needle; // both -1 when nothing was found
+            alias steps this; // compatible with previous ptrdiff_t return type
+            ptrdiff_t opIndex(size_t idx) // faking a tuple
```

Comment:
> I'm playing with the idea to change the signature to ptrdiff_t opIndex(bool idx). Due to implicit conversion, res[0] and res[1] would still work. 

Yeah that's quite weird. Imho - if we want to improve things - we should look into how we could trigger deprecation warnings if the `alias steps this` is used, but I'm not sure whether that's possible.

>  tuples do not satisfy isRandomAccessRange

This emulation isn't even an `InputRange`.